syntax = "proto3";

package supervisor;

service Supervisor {
  rpc start_pipeline(StartPipelineRequest) returns (PipelineResponse) {}
  rpc get_pipeline_status(GetPipelineStatusRequest) returns (GetPipelineStatusResponse) {}
}

message JsonString { string value = 1; }

message StartPipelineRequest {
    JsonString pipeline_config = 1;
    string eval_directory = 2;
    optional int32 start_replay_at = 3;
    optional int32 stop_replay_at = 4;
    optional int32 maximum_triggers = 5;
}

message PipelineResponse { int32 pipeline_id = 1; }

message GetPipelineStatusRequest { int32 pipeline_id = 1; }

message GetPipelineStatusResponse {
    string status = 1;
    JsonString detail = 2;
}